### PR TITLE
Finish user singup

### PR DESCRIPTION
**Pull request by phucld95**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ git_source(:github) do |repo_name|
 end
 
 gem "rails", "~> 5.0.1"
+gem 'bcrypt', "3.1.11"
```
Comment (vietmt) [inline Gemfile]: Nháy kép

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,4 +1,29 @@
 class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+    unless @user
+    flash[:danger] = "User not exist"
```
Comment (vietmt) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
+      redirect_to @user
+    else
+      render "new"
```
Comment (vietmt) [inline app/controllers/users_controller.rb]: render :new

File: app/views/shared/_error_messages.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize @user.errors.count, "error" %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (vietmt) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize @user.errors.count, "error" %>.
+    </div>
+    <ul>
+      <% @user.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
```
Comment (DuongNT1512) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -1,2 +1,18 @@
...
+<h1>Sign up</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user, url: signup_path do |f| %>
+      <%= render "shared/error_messages" %>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :email %>
+      <%= f.email_field :email, class: "form-control" %>
+	  <%= f.label :password %>
```
Comment (DuongNT1512) [inline app/views/users/new.html.erb]: indent

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,4 +1,29 @@
...
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
+      redirect_to @user
+    else
+      render :new
+    end
   end
+
+  private
+    def user_params
```
Comment (hellovietnam93) [inline app/controllers/users_controller.rb]: ident, thằng hàng vs cái priavate nhé

**Pull request merged**